protocol/bc/asset.go

```diff
@@ -67,5 +67,8 @@ func (a AssetAmount) WriteTo(w io.Writer) (int64, error) {
 }
 
 func (a AssetAmount) Equal(other *AssetAmount) bool {
+	if a.AssetId == nil || other.AssetId == nil {
+		return false
```

Comment:
What if they're both nil? Should it return `a.AssetId == nil && other.AssetId == nil` to cover that case? Or should that case be false too.